pyscal/utils/string.py

```diff
@@ -2,20 +2,20 @@
 
 import logging
 
+import pandas as pd
+
 from .monotonicity import modify_dframe_monotonicity
 
 logger = logging.getLogger(__name__)
 
 
 def df2str(
-    dframe,
-    digits=7,
-    roundlevel=9,
-    header=False,
-    monotonicity=None,
-    monotone_column=None,
-    monotone_direction=None,
-):
+    dframe: pd.DataFrame,
+    digits: int = 7,
+    roundlevel: int = 9,
+    header: bool = False,
+    monotonicity: dict = None,
```

Comment:
`Optional[dict]`?

If you want to include more information regarding input here (and maybe other cases?), looking at
> monotonicity: Settings for monotonicity in output. A dict
            with column names as keys, with values being a dict with keys
            "sign" (-1 or +1 integer) for direction,  "upper" and "lower" for
            lower and upper limits (non-strict monotonicity is allowed at
            these upper and lower limits).

it might be a good use case for https://www.python.org/dev/peps/pep-0589/ (we use it successfully in `webviz-config`). Then it becomes clear directly from the code which keys to include (and `mypy` will catch if consumers of the API use other/missing dict keys).

E.g. something like `Optional[Dict[str, MonotonicityDict]]` if I understand the docstring correctly?